b2sdk/utils/thread_pool.py

```diff
@@ -0,0 +1,55 @@
+######################################################################
+#
+# File: b2sdk/utils/thread_pool.py
+#
+# Copyright 2022 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+from concurrent.futures import ThreadPoolExecutor
+from threading import Lock
+from typing import Optional
+
+from b2sdk.utils import B2TraceMetaAbstract
+
+
+class LazyThreadPoolMixin(metaclass=B2TraceMetaAbstract):
+    """
+    Mixing class with lazy initialized ThreadPoolExecutor.
+    Can be safely used between threads.
+    """
+
+    def __init__(self, max_workers: 'Optional[int]' = None, **kwargs) -> None:
+        """
+        :param Optional[int] max_workers: maximum number of worker threads
+        """
+
+        self._lock = Lock()
+        self._thread_pool = None  # type: 'Optional[ThreadPoolExecutor]'
+        self._max_workers = max_workers
+        super().__init__(**kwargs)
+
+    def _set_thread_pool_size(self, max_workers: int) -> None:
```

Comment:
what's the point of this method? Is it here only for the sake of providing a public version of it in spicer wrappers?

Reply:
Yes, we agreed with @ppolewicz  that it is bad design to expose it, but for compatibility reasons we decided to keep it as protected.

Reply:
in that case, I don't think it should be here. If it's only purpose is to be used in spicer wrappers, then it should be defined there

Reply:
yes, those should be moved to `b2sdk/v2`

Reply:
Already done.